### DISCUSSION TITLE
Support for note-taking

### DISCUSSION THREAD

**Discussion by gzko** [sumatrapdfreader/sumatrapdf#3481] [Ideas]
It could be convenient for the selection handler to get $(file)
 for file name (url-encoded path) and $(page) for page number
in addition to $(selection)?

**Comment by GitHubRulesOK** [comment]
HMM those are seperate uses  in that $selection is one web use but "%f %l" or %1 %p are for editors

Filename and page were always there for TeX Enhancements so double click would open the TeX Editor with file and current line when a synctex or pdfsync file  is present. 
It is in theory possible to craft your own pdfsync for each PDF, but I would not recommend that.

For calling current page in Acrobat Pro, MS NotePad or Write OR Tracker Xchange PDF editor or any programmable Command Line tool etc there is "ExternalViewers" but that includes Shell page number to Editors as Editors are Viewers too, what is missing is that call excludes selection.

**Reply by gzko** [reply]
It is for saving a simple note with the appropriate software.
One click, just file name, page, selected text (and creation date).
Let's say a plain text or markdown file in the provided folder.

Sumatra PDF already supports: /A page=%p "%1", to use that.

**Comment by GitHubRulesOK** [comment]
I gave examples on old forum see https://forum.sumatrapdfreader.org/t/advanced-tips-add-your-own-notes/2688

In case that is down
here was the first simplistic step (later its changed to use a more powerful script with WordPad or other editors)
at this point, it shows a  very basic ONE.TXT file PER page that due to syntax changes old forum note does not work well with notepad so here is my current one.

```
ExternalViewers [
  	[
		CommandLine = "C:\windows\system32\cmd.exe" /c do edit "%1" "-Page%p.txt"
		Name = Notepad the Notes for &This page
		Filter = *.*
	]
  ]
```
the external "EDITOR" in do.bat is much more than this but this is the edit bit
```
:Edit
if %2.==. Start "EditDo" Notepad "%~f0"
if not %2.==. Start "EditOR" Notepad "%~dpn2%~3"
exit /b 0
goto error Edit
```

![image](https://user-images.githubusercontent.com/12754913/236488470-4d3af3b0-21d1-428a-8211-114b716e0b1b.png)


Mine is a bit more "ADVANCED" so will look like this

![image](https://user-images.githubusercontent.com/12754913/236503803-5ce85747-0ca5-4788-b111-c241b2c1718f.png)

The SumatraPDF call has been simplified to
```
	[
		CommandLine = "C:\windows\system32\cmd.exe" /c do edit "%1" %p
		Name = Notepad the Notes for &This page
		Filter = *.*
	]
```

but my system wide do.bat has this section (beware this is for illustration as has no error checks) You need to write your own tasks.cmd


![image](https://user-images.githubusercontent.com/12754913/236505275-34ccb73e-66d3-48be-8bfa-b06f511540aa.png)

**Reply by gzko** [reply]
Yes, I had something like that in mind. But automated, with minimal support.

**Comment by GitHubRulesOK** [comment]
Automated what ??  that's three keys to send page content to writer and my support is free

![image](https://user-images.githubusercontent.com/12754913/236518056-aa9549b2-1470-492d-8830-aa51c6c1c511.png)
